imap/monitor.go

```diff
@@ -192,3 +200,45 @@ func checkForNewEmails(im models.IMAP) {
 		log.Debug("No new emails for ", im.Username)
 	}
 }
+
+// returns a slice of gophish rid paramters found in the email HTML, Text, and attachments
+func checkRIDs(em *email.Email) (map[string]int, error) {
```

Comment:
What's the use case you see where more than one `rid` can be reported per email? I can't really see that happening, but I could be missing something.

If we don't anticipate that happening, this might be a bit easier to implement.

Reply:
So I was trying to cover as many eventualities as possible - perhaps an email has an rid parameter that's not from gophish (some other developer created links with the same rid parameter name/form), or someone forwards two different campaign emails in one.

Happy to remove and simplify if you'd prefer though. We could change the function to singular `checkRID` and just return the first one it finds.